constellations/gemini-user/src/programs/programModal.js

```diff
@@ -0,0 +1,48 @@
+"use strict";
+require("./programModal.sass");
+import React from "react";
+import { Link } from "react-router-dom";
+
+export class ProgramModal extends React.Component {
+    render = () => {
+        const semester = this.props.semester;
+        const classes = this.props.classes;
+
+        const classList = classes.map((c, index) => (
+            <ProgramClass key={index} classObj={c} />
+        ));
+
+        return (
+            <div className="program-modal">
+                <h1>{semester.title + " Classes"}</h1>
+                <ul>{classList}</ul>
+            </div>
+        );
+    };
+}
+
+class ProgramClass extends React.Component {
+    getTimes = (times) => {
+        return times
+            .split(",")
+            .map((time, index) => <span key={index}>{time.trim()}</span>);
+    };
+
+    render = () => {
+        const classObj = this.props.classObj;
+        const name =
+            classObj.classKey +
+            ["", " (ALMOST FULL)", " (FULL)"][classObj.fullState];
```

Comment:
Can you move this list to a `constants.js` file in the utils folder? I think you might need to also include this information in the ClassPage. Maybe place it right above the "Grades" part? And if it's full or almost-full, make the font red.

And maybe name it like this: `FULL_STATE_DISPLAY_NAMES`. For "global" constants, we like to use upper-capitalized snake_case.

```
FULL_STATE_DISPLAY_NAMES = ["", " (ALMOST FULL)", " (FULL)"]
```